shared/src/api/QuestionGenerator.ts

```diff
@@ -195,6 +195,9 @@ export type FreeTextFormatFunction = (
  * example MultipleChoiceQuestion, etc.
  */
 export interface QuestionGenerator {
+  /** Unique and stable identifier of the question generator. */
+  id: string
```

Comment:
This id is used directly in the URLs. So it should be documented that this is the case and maybe what values we expect here. E.g. it cannot contain `/` and some other characters. maybe we want to have them short. But as it looks the currently have a word or an abbreviation and no hash or something like that.

Reply:
This was only documented in the unit test, and I added a comment here to explain the currently rather strict rules.